lib/portlayer/storage/vsphere/image.go

```diff
@@ -288,14 +288,14 @@ func (v *ImageStore) Export(op trace.Operation, store *url.URL, id, ancestor str
 	mounts := []*object.DatastorePath{}
 	cleanFunc := func() {
 		for _, mount := range mounts {
-			if err := v.dm.UnmountAndDetach(op, mount); err != nil {
+			if err := v.dm.UnmountAndDetach(op, mount, !persistent); err != nil {
```

Comment:
Just change the UnmountAndDetach to accept VirtualDisk and use DiskMode internally in that function to decide what to do https://github.com/vmware/vic/blob/master/pkg/vsphere/disk/config.go#L39